crates/transaction-emitter-lib/src/emitter/transaction_executor.rs

```diff
@@ -79,9 +132,10 @@ async fn submit_and_check(
     rest_client: &RestClient,
     txn: &SignedTransaction,
     wait_duration: Duration,
-    submit_failure_counter: &AtomicUsize,
-    wait_failure_counter: &AtomicUsize,
+    failed_submit: &mut bool,
+    failed_wait: &mut bool,
```

Comment:
this is a bit of anti-pattern, typically it should be part of the Err enum variants but anyway

Reply:
I know, but failed_submit can be true, while failed_wait is false and it returns Ok, so I would need to create a new enum for Either (because rust doesn't have it), etc ,etc etc. so doing this :)